src/components/EquippedCharacterSheetSlot/index.tsx

```diff
@@ -0,0 +1,106 @@
+import styled from "styled-components";
+import { AttributeImpact, CharacterSheetSlot, SlotType } from "../../types";
+import activeSpellPath from "./active_spell_icon.jpg";
+import instantSpellPath from "./instant_spell_icon.jpg";
+import weaponPath from "./weapon_icon.jpg";
+import Icon from "../Icon";
+import RenderedDice from "../RenderedDice";
+import { DiceAttribute } from "../../attributes";
+
+export interface Props {
+  slot: CharacterSheetSlot;
+}
+
+interface AttributeProps {
+  attributesImpacted: AttributeImpact[];
+}
+
+interface SlotIconsProps {
+  slotTypes: SlotType[];
+}
+
+const getIconPath = (powerType: SlotType) => {
+  if (powerType === "active") {
+    return activeSpellPath;
+  } else if (powerType === "instant") {
+    return instantSpellPath;
+  } else {
+    return weaponPath;
+  }
+};
+
+const EquippedCharacterSheetSlot = (props: Props) => {
+  return (
+    <Container>
+      <SlotIcons slotTypes={props.slot.slotTypes} />
+      <MainContent>
+        <Icon path={props.slot.iconLink} height={32} width={32} />
+        <CostBox>{props.slot.energyCost}</CostBox>
+        <div style={{ width: "100px" }}> {props.slot.name} </div>
+        <AttributesImpactedView attributesImpacted={props.slot.attributesImpacted} />
+      </MainContent>
+    </Container>
+  );
+};
+
+const AttributesImpactedView = (props: AttributeProps) => {
+  return (
+    <AttributesContainer>
+      {props.attributesImpacted
+        .filter((attr) => attr.attribute.name === "dice")
+        .sort()
+        .map((attr) => {
+          const dice = attr.attribute as DiceAttribute;
```

Comment:
as you may know, we want to avoid this kind of dangerous cast as much as possible (ideally never do this).

typescript should be able to infer that `attribute` is a `DiceAttribute` when doing `if (attribute.name === "dice")`. I think you can do this by either directly typing `DiceAttribute.name` exactly as `"dice"`, or making `Attribute` a union type of objects representing all the different attribute types (all with hardcoded names as well)